### PR TITLE
feat: add stress tests

### PR DESCRIPTION
**Pull request by x-Milosz**
Added stress testś that will error out if map with 100k elements loads below 20 seconds.

See: https://github.com/Problematy/goodmap/issues/189

## Summary by CodeRabbit

- **New Features**
	- Introduced end-to-end stress testing capabilities, including data generation and test execution.
	- Added a new test suite to measure application performance under stress.
	- New configuration file for stress testing with application settings and language support.

- **Bug Fixes**
	- Updated testing workflow to include steps for generating stress test data and running stress tests.

- **Chores**
	- Added entries to `.gitignore` to exclude generated test data from version control.
	- Modified the `Makefile` to streamline testing processes.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -48,3 +48,11 @@ jobs:
         PYTHONUNBUFFERED=1;FLASK_ENV=development;FLASK_APP="goodmap.goodmap:create_app(config_path='./tests/e2e_tests/e2e_test_config.yml')" poetry run flask run &
         sleep 3
         cd tests/e2e_tests && node_modules/cypress/bin/cypress run --browser chromium
+    - name: Generate e2e stress test data
+      run: |
+        python tests/e2e_tests/cypress/support/generate_stress_test_data.py
+    - name: Run e2e stress tests
```
[the remark this answers is not in the document]
Comment (raven-wing) [reply .github/workflows/tests.yml:56]: If generating test data fails then test will fail none the less.. it's not needed

File: tests/e2e_tests/e2e_stress_test_config.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+APP_NAME: Bridges in Wroclaw
+SECRET_KEY: SECRET
```
[the remark this answers is not in the document]
Comment (raven-wing) [reply tests/e2e_tests/e2e_stress_test_config.yml:2]: That's test's secret key so it doesn't matter (but keep searching for those)

File: tests/e2e_tests/e2e_stress_test_config.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    name: English
+    flag: gb
+    country: GB
+  pl:
+    name: polski
+    flag: pl
+    country: PL
+
+FEATURE_FLAGS:
+  USE_LAZY_LOADING: False
```
Comment (raven-wing) [inline tests/e2e_tests/e2e_stress_test_config.yml:20]:
```suggestion
  USE_LAZY_LOADING: True
```

Let's check it with faster code. :)

Please update timer acrodingly.

File: tests/e2e_tests/cypress/support/generate_stress_test_data.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
+import json
+import random
+import uuid
+
+
+def generate_random_position_within_poland():
```
Comment (raven-wing) [inline tests/e2e_tests/cypress/support/generate_stress_test_data.py:6]: Ok. I had to think that through... although I like randomness approach it probably is not best to have it in stress test. Let's keep same dataset for every run to be sure we have the same condition.

**Review by raven-wing — CHANGES_REQUESTED** [review]

File: Makefile [hunk 5]
```diff
@@ -17,7 +17,13 @@ unit-tests-no-coverage:
 	poetry run python -m pytest -m "skip_coverage"
 
 e2e-tests:
-	cd tests/e2e_tests && node_modules/cypress/bin/cypress run --browser chromium
+	cd tests/e2e_tests && node_modules/cypress/bin/cypress run --browser chromium --spec cypress/e2e/basic-test/*
+
+e2e-stress-tests-generate-data:
+	python tests/e2e_tests/cypress/support/generate_stress_test_data.py
+
```
[the remark this answers is not in the document]
Comment (raven-wing) [reply Makefile:24]: Will be done in issue #198

File: tests/e2e_tests/cypress/e2e/stress-test/stress_test.cy.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+      cy.window().then((win) => {
+        let startTime = win.performance.now();
+
+        cy.get('#map', { timeout: 60000 }).children({ timeout: 60000 })
+          .then(() => {
+            cy.window().then((win) => {
+              let endTime = win.performance.now();
+              const runTime = endTime - startTime;
+              cy.log(`Run ${runTimes.length + 1} took ${runTime}ms`);
```
Comment (raven-wing) [inline tests/e2e_tests/cypress/e2e/stress-test/stress_test.cy.js:21]: Can we somehow store this for future runs to compare results?

File: tests/e2e_tests/cypress/e2e/stress-test/stress_test.cy.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+describe("Stress test", () => {
+
+  it('Should map stop loading and measure performance', function () {
+    let slowestTime = 0;
+    const numRuns = 5;
+    let runTimes = [];
```
[the remark this answers is not in the document]
Comment (raven-wing) [reply tests/e2e_tests/cypress/e2e/stress-test/stress_test.cy.js:6]: referenced in #205

[the remark this answers is not in the document]
Comment (raven-wing) [reply tests/e2e_tests/cypress/e2e/stress-test/stress_test.cy.js:6]: referenced in #206

File: tests/e2e_tests/cypress/e2e/stress-test/stress_test.cy.js (hunk 6 above)
Comment (raven-wing) [reply tests/e2e_tests/cypress/e2e/stress-test/stress_test.cy.js:21]: referenced in #207

File: tests/e2e_tests/cypress/support/generate_stress_test_data.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        "name": generate_random_name(),
+        "position": generate_random_position_within_poland(),
+        "accessible_by": generate_random_accessible_by(),
+        "type_of_place": generate_random_type_of_place(),
+        "UUID": str(uuid.uuid4()),
+    }
+
+
+def generate_data_set(num_items):
+    return [generate_data_item() for _ in range(num_items)]
```
[the remark this answers is not in the document]
Comment (raven-wing) [reply tests/e2e_tests/cypress/support/generate_stress_test_data.py:38]: It's already using generator isnt it?

File: tests/e2e_tests/cypress/support/generate_stress_test_data.py [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+        "secondary_color": "#245466",
+        "left_bar_width": "300px",
+    },
+    "plugins": [
+        {
+            "name": "redirections",
+            "config": {
+                "/test": "/blog/page/about",
+                "/static/map.js": "https://cdn.jsdelivr.net/npm/@problematy/goodmap@0.3.5",
+            },
```
[the remark this answers is not in the document]
Comment (raven-wing) [reply tests/e2e_tests/cypress/support/generate_stress_test_data.py:101]: It's good idea, but for now we do not have solution for that.

[the remark this answers is not in the document]
Comment (raven-wing) [reply tests/e2e_tests/cypress/support/generate_stress_test_data.py:101]: In #208

File: tests/e2e_tests/cypress/support/generate_stress_test_data.py (hunk 4 above)
Comment (x-Milosz) [reply tests/e2e_tests/cypress/support/generate_stress_test_data.py:6]: Set seed. Now points will always generate in the same way

File: tests/e2e_tests/e2e_stress_test_config.yml (hunk 3 above)
Comment (x-Milosz) [reply tests/e2e_tests/e2e_stress_test_config.yml]: Done

**Review by raven-wing — APPROVED** [review]

**Pull request merged**